third_party/2and3/attr/__init__.pyi

```diff
@@ -21,7 +21,13 @@ _ValidatorArgType = Union[_ValidatorType[_T], Sequence[_ValidatorType[_T]]]
 NOTHING: object
 
 # NOTE: Factory lies about its return type to make this possible: `x: List[int] = Factory(list)`
-def Factory(factory: Union[Callable[[], _T], Callable[[Any], _T]], takes_self: bool = ...) -> _T: ...
+# Work around mypy issue #4554 by using an overload vs a Union.
+@overload
+def Factory(factory: Callable[[], _T]) -> _T: ...
```

Comment:
Isn't this equivalent to the second overload? If not why, don't you need one where the Callable takes an argument and the second argument is omitted?

Reply:
If you don't set `takes_self` then you'd better not send in a function with an argument.  If you do set `takes_self` then both are allowed.

That being said I'm ok with switching this to:

```
@overload
def Factory(factory: Callable[[], _T], takes_self: bool = ...) -> _T: ...
@overload
def Factory(factory: Callable[[Any], _T], takes_self: bool = ...) -> _T: ...
```
And forget about the first overload.

Reply:
Yes, the first overload is redundant because the argument has a default.